### PR TITLE
Add conversion of poly polynomial to cvc node.

### PR DESCRIPTION
**Pull request by nafur**
This PR adds a new utility function to convert a `poly::Polynomial` back to a cvc4 Node.

### CODE REVIEW

File: src/theory/arith/nl/poly_conversion.cpp [hunk 1]
```diff
@@ -211,6 +211,67 @@ poly::Polynomial as_poly_polynomial(const CVC4::Node& n,
...
 }
 
+namespace {
+/**
+ * Utility class that collects the monomial terms (as nodes) from the polynomial
+ * we are converting.
+ */
+struct CollectMonomialData
+{
+  VariableMapper& vm;
```
Comment (ajreynol) [inline src/theory/arith/nl/poly_conversion.cpp:221]: Remember to use `d_` for member data.

File: src/theory/arith/nl/poly_conversion.cpp [hunk 2]
```diff
@@ -211,6 +211,67 @@ poly::Polynomial as_poly_polynomial(const CVC4::Node& n,
...
+  std::vector<Node> terms;
+  NodeManager* nm = NodeManager::currentNM();
+
+  CollectMonomialData(VariableMapper& v) : vm(v) {}
```
Comment (ajreynol) [inline src/theory/arith/nl/poly_conversion.cpp:225]: This should be at the top of the struct.

**Review by ajreynol — CHANGES_REQUESTED** [review]
A few style issues.

File: src/theory/arith/nl/poly_conversion.cpp [hunk 3]
```diff
@@ -211,6 +211,67 @@ poly::Polynomial as_poly_polynomial(const CVC4::Node& n,
...
+namespace {
+/**
+ * Utility class that collects the monomial terms (as nodes) from the polynomial
+ * we are converting.
+ */
+struct CollectMonomialData
+{
+  CollectMonomialData(VariableMapper& v) : d_vm(v) {}
+
+  VariableMapper& d_vm;
```
Comment (ajreynol) [inline src/theory/arith/nl/poly_conversion.cpp:223]: Can you give one line doc for these?

Comment (nafur) [reply src/theory/arith/nl/poly_conversion.cpp:224]: Done.

**Review by ajreynol — APPROVED** [review]
LGTM

**Pull request merged**